lib/blacklight/configuration.rb

```diff
@@ -288,15 +276,12 @@ def deep_copy
     end
     alias_method :inheritable_copy, :deep_copy
 
-    ##
     # Get a view configuration for the given view type
     # including default values from the index configuration
+    # @param [Symbol,#to_sym] view_type
     def view_config(view_type)
-      if view_type == :show
-        self.index.merge self.show
-      else
-        self.index.merge view.fetch(view_type, {})
-      end
+      view_type = view_type.to_sym unless view_type.is_a? Symbol
```

Comment:
@mejackreed Yes, it is is quite plausible that it is (already) required elsewhere, too. This PR addresses the concrete actual problem in #1587.  I support doing it at a lower level **also**. But that is not necessary to resolve 1587 AND is insufficient to allow consumers of `view_config` to use symbols and strings interchangeably.  Ticket that separately if you want me to work on it.

As I said in response to Chris, we cannot *only* do `to_sym` at the lower level if you want the comparison in the very next line to still be valid for both string and symbol arguments, that is to say, specifically, if we want to treat these lines as equivalent: 
```ruby
view_config('show')
view_config(:show)
```

I hear you saying that you want that: this provides it.